tee-worker/litentry/core/evm-dynamic-assertions/src/lib.rs

```diff
@@ -66,6 +71,10 @@ pub type AssertionParams = Vec<u8>;
 pub type SmartContractByteCode = Vec<u8>;
 pub type AssertionRepositoryItem = (SmartContractByteCode, Vec<String>);
 
+thread_local! {
+	pub static DYNAMIC_ASSERTION_LOGS: RefCell<Vec<String>> = RefCell::new(Vec::new());
```

Comment:
Could you extend `struct Precompiles` with logs vector - it's created for each smart contract execution and you  can access it inside precompile (if you pass it) and after execution you can just return logs stored in Precompiles struct?

This way we can avoid using thread local variables - we are sure no data is shared between executions